From c782e208021409e9b78acb2200abd4319072e78a Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Fri, 2 Sep 2022 00:28:05 -0700
Subject: [PATCH] Fix type of the void pointer assignment

Fixes build with clang

x86-common.c:216:9: error: incompatible integer to pointer conversion assigning to 'void *' from 'long' [-Wint-conversion]
        offset = mem_info.offset - REAL_MEM_BASE;
               ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Upstream-Status: Pending
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 x86-common.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/x86-common.c b/x86-common.c
index 137bc3c..6f737ed 100644
--- a/x86-common.c
+++ b/x86-common.c
@@ -213,7 +213,7 @@ void *LRMI_common_init(int high_page)
 	if (!real_mem_init(high_page))
 		return NULL;
 
-	offset = mem_info.offset - REAL_MEM_BASE;
+	offset = (void*)(mem_info.offset - REAL_MEM_BASE);
 
 	/*
 	 Map the Interrupt Vectors (0x0 - 0x400) + BIOS data (0x400 - 0x502)
-- 
2.37.3

